main.go

```diff
@@ -1,426 +0,0 @@
-	"flag"
-	"io/ioutil"
-	"net"
-	"net/http"
-	"net/http/httputil"
-	"net/url"
-	"os"
-	"os/signal"
-	"path"
-	"strings"
-	"syscall"
-	"time"
-
-	"github.com/ghodss/yaml"
-	"github.com/oklog/run"
-	"github.com/spf13/pflag"
-	"golang.org/x/net/http2"
-	"golang.org/x/net/http2/h2c"
-	"k8s.io/apiserver/pkg/authentication/authenticator"
-	"k8s.io/apiserver/pkg/authorization/union"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
-	certutil "k8s.io/client-go/util/cert"
-	k8sapiflag "k8s.io/component-base/cli/flag"
-	"k8s.io/klog/v2"
-
-	"github.com/brancz/kube-rbac-proxy/pkg/authn"
-	"github.com/brancz/kube-rbac-proxy/pkg/authz"
-	"github.com/brancz/kube-rbac-proxy/pkg/proxy"
-	rbac_proxy_tls "github.com/brancz/kube-rbac-proxy/pkg/tls"
-)
-
-type config struct {
-	insecureListenAddress string
-	secureListenAddress   string
-	upstream              string
-	upstreamForceH2C      bool
-	upstreamCAFile        string
-	auth                  proxy.Config
-	tls                   tlsConfig
-	kubeconfigLocation    string
-	allowPaths            []string
-	ignorePaths           []string
-}
-
-type tlsConfig struct {
-	certFile       string
-	keyFile        string
-	minVersion     string
-	cipherSuites   []string
-	reloadInterval time.Duration
-}
-
-type configfile struct {
-	AuthorizationConfig *authz.Config `json:"authorization,omitempty"`
-}
-
-func main() {
-	cfg := config{
-		auth: proxy.Config{
-			Authentication: &authn.AuthnConfig{
-				X509:   &authn.X509Config{},
-				Header: &authn.AuthnHeaderConfig{},
-				OIDC:   &authn.OIDCConfig{},
-				Token:  &authn.TokenConfig{},
-			},
-			Authorization: &authz.Config{},
-		},
-	}
-	configFileName := ""
-
-	// Add klog flags
-	klogFlags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
-	klog.InitFlags(klogFlags)
-
-	flagset := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
-	flagset.AddGoFlagSet(klogFlags)
-
-	// kube-rbac-proxy flags
-	flagset.StringVar(&cfg.insecureListenAddress, "insecure-listen-address", "", "The address the kube-rbac-proxy HTTP server should listen on.")
-	flagset.StringVar(&cfg.secureListenAddress, "secure-listen-address", "", "The address the kube-rbac-proxy HTTPs server should listen on.")
-	flagset.StringVar(&cfg.upstream, "upstream", "", "The upstream URL to proxy to once requests have successfully been authenticated and authorized.")
-	flagset.BoolVar(&cfg.upstreamForceH2C, "upstream-force-h2c", false, "Force h2c to communiate with the upstream. This is required when the upstream speaks h2c(http/2 cleartext - insecure variant of http/2) only. For example, go-grpc server in the insecure mode, such as helm's tiller w/o TLS, speaks h2c only")
-	flagset.StringVar(&cfg.upstreamCAFile, "upstream-ca-file", "", "The CA the upstream uses for TLS connection. This is required when the upstream uses TLS and its own CA certificate")
-	flagset.StringVar(&configFileName, "config-file", "", "Configuration file to configure kube-rbac-proxy.")
```

Comment:
This is the authorization config, distinct from the authentication config?

Reply:
Yes, the config file is authz only. Authentication is handled by flags. Might have grown organically in time.

The pre-acceptance task is to update documentation in CLI and README, not the uniform handling of configuration options, right?

Reply:
> The pre-acceptance task is to update documentation in CLI and README, not the uniform handling of configuration options, right?

correct.  Ideally with links to where I can see what format I've got here.  I couldn't tell what this was without reading the code.